dcm4che-core/src/main/java/org/dcm4che3/data/Attributes.java

```diff
@@ -1951,8 +1961,10 @@ public Fragments newFragments(String privateCreator, int tag, VR vr,
     }
 
     private Object set(String privateCreator, int tag, VR vr, Object value) {
-        if (vr == null)
-            throw new NullPointerException("vr");
+        if (vr == null) {
```

Comment:
The file I was testing with had a 0 encoded instead of an UN code, which caused vr=null here.

Reply:
I think this shouldn't be part of this commit. If you want to handle that more lenient, I would suggest to create a separate fix and also open that as a pull request for the master branch.